jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c

```diff
@@ -623,114 +726,67 @@ ecma_builtin_helper_string_prototype_object_index_of (ecma_string_t *original_st
  * This function clamps the given index to the [0, length] range.
  * If the index is negative, 0 value is used.
  * If the index is greater than the length of the string, the normalized index will be the length of the string.
- * NaN is mapped to zero or length depending on the nan_to_zero parameter.
  *
  * See also:
- *          ECMA-262 v5, 15.5.4.7,8,11
+ *          ECMA-262 v6, 21.1.3.8
  *
  * Used by:
  *         - The ecma_builtin_helper_string_prototype_object_index_of helper routine.
  *         - The ecma_builtin_string_prototype_object_replace_match helper routine.
  *
- * @return bool - whether there is a match for the search string
+ * @return uint32_t - whether there is a match for the search string
  */
-bool
+uint32_t
 ecma_builtin_helper_string_find_index (ecma_string_t *original_str_p, /**< index */
                                        ecma_string_t *search_str_p, /**< string's length */
-                                       bool first_index, /**< whether search for first (t) or last (f) index */
                                        ecma_length_t start_pos, /**< start position */
                                        ecma_length_t *ret_index_p) /**< [out] position found in original string */
 {
-  bool match_found = false;
-  const ecma_length_t original_len = ecma_string_get_length (original_str_p);
-  const ecma_length_t search_len = ecma_string_get_length (search_str_p);
-
-  if (search_len <= original_len)
+  uint32_t match_found = ECMA_VALUE_EMPTY;
+  if (ecma_string_is_empty (search_str_p))
   {
-    if (!search_len)
-    {
-      match_found = true;
-      *ret_index_p = first_index ? start_pos : original_len;
-    }
-    else
-    {
-      /* create utf8 string from original string and advance to position */
-      ECMA_STRING_TO_UTF8_STRING (original_str_p, original_str_utf8_p, original_str_size);
-
-      ecma_length_t index = start_pos;
-
-      const lit_utf8_byte_t *original_str_curr_p = original_str_utf8_p;
-      for (ecma_length_t idx = 0; idx < index; idx++)
-      {
-        lit_utf8_incr (&original_str_curr_p);
-      }
-
-      /* create utf8 string from search string */
-      ECMA_STRING_TO_UTF8_STRING (search_str_p, search_str_utf8_p, search_str_size);
+    match_found = ecma_make_uint32_value (start_pos);
+    *ret_index_p = start_pos;
```

Comment:
`ret_index_p` no longer seems to be necessary, the returned valus has the same information.